### PR TITLE
add node-fetch based on whatwg-fetch (#8931)

### PR DESCRIPTION
**Pull request by torstenwerner**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

It is primarily a copy of whatwg-fetch with the idea from #8931 applied.

### CODE REVIEW

**Comment by torstenwerner** [comment]
I have added my fork to travis yesterday but travis does not see any requests coming in: https://travis-ci.org/torstenwerner/DefinitelyTyped/requests . I have pushed a comment change, created another branch, and clicked the 'test service' button at github but there is still no build. Other unrelated repositories are still building successfully. Do you have any ideas on how to get this repo built?

**Comment by torstenwerner** [comment]
- [x] fixed comments in node-fetch.d.ts
- [x] pass the Travis CI test https://travis-ci.org/torstenwerner/DefinitelyTyped

File: node-fetch/node-fetch.d.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	status: number;
+	statusText?: string;
+	headers?: HeaderInit;
+}
+
+declare type HeaderInit = Headers|Array<string>;
+declare type BodyInit = ArrayBuffer|ArrayBufferView|Blob|FormData|string;
+declare type RequestInfo = Request|string;
+
+declare module 'node-fetch' {
```
Comment (vvakame) [inline node-fetch/node-fetch.d.ts]: suggestion. move this line to L6.
We should reduce exported symbol for global namespace.

**Comment by torstenwerner** [comment]
I have tried to move the line 88 up to 6 but the tests fail now, e.g.
`node-fetch/node-fetch-tests.ts(6,20): error TS2304: Cannot find name 'Headers'.`
What is the best way to fix this?

**Comment by vvakame** [comment]
@torstenwerner I read node-fetch's source code.
https://github.com/bitinn/node-fetch/blob/master/index.js#L224-L227
node-fetch is not modify global object that mean We should not use `Headers` and `Request` object from global namespace. isn't it?

**Comment by torstenwerner** [comment]
@vvakame I do not object changing the node-fetch.d.ts file but all the tests are broken now. How do i fix them e.g. the line
`var request: Request = new Request("http://www.andlabs.net/html5/uCOR.php", requestOptions);`
that references the types in the .d.ts file. Should I copy most of the types into the test file what would be very non DRY? Is there a better way to handle the issue? Maybe I am not aware of the right solution?

**Comment by torstenwerner** [comment]
@vvakame I think I could fix it now. Could you please have a look again?

**Comment by vvakame** [comment]
I think this code is valid in TypeScript & compiled JavaScript.

```
import fetch = require('node-fetch');

function test_fetchUrlWithOptions() {
	var headers = new fetch.Headers();
	headers.append("Content-Type", "application/json");
	var requestOptions: _fetch.RequestInit = {
		method: "POST",
		headers: headers,
		mode: 'same-origin',
		credentials: 'omit',
		cache: 'default',
		redirect: 'manual'
	};
	handlePromise(fetch("http://www.andlabs.net/html5/uCOR.php", requestOptions));
}
```

---

https://tonicdev.com/npm/node-fetch I tried your test code in this URL.
```
var fetch = require('node-fetch');
var headers = new _fetch.Headers();
```

this code returns errors. `ReferenceError: _fetch is not defined`

```
var fetch = require('node-fetch');
var headers = new Headers();
```

this code returns errors. `ReferenceError: Headers is not defined`

```
var fetch = require('node-fetch');
var headers = new fetch.Headers();
```

works fine!

**Comment by mhegazy** [comment]
looks like @vvakame signed off. merging.

**Pull request merged**